### ISSUE TITLE
Error: failed to set EPP value

### ISSUE THREAD

**Issue by LeiZhou-97** [intel/kubernetes-power-manager#15]
I followed the instruction to deploy the kubernetes-power-manager and then I applied an example-power-workload.yaml. But It didn't set EPP value(powerprofile) correctly. I logged into the pod and tried to set EPP value manually, it also showed the same error message. 

**Error log:** 
failed to set EPP value for core 0: open /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_preference: **read-only file system**

**Docker image:** intel/power-node-agent:v2.0.0

**Comment by patricia-cahill** [comment]
Hey, 
Thanks a million for reaching out to us.  
Regarding the workload, CPUs 0 and 1 are reserved CPUs so the Power Manager wont be doing any configuration with them.
Also have you imputed the correct node name into the placeholder?

**Comment by LeiZhou-97** [comment]
Hi,
Thanks for such a quick reply！
This is my powerworkload yaml and I've replaced <my-node-name> to my real node name. Based on your suggestion, I excluded the CPU0and1, but I still got the same err logs.
**Error log:**
failed to set EPP value for **core 2:** open /sys/devices/system/cpu/**cpu2**/cpufreq/energy_performance_preference: read-only file system
```
apiVersion: power.intel.com/v1
kind: PowerWorkload
metadata:
  name: performance-<my-node-name>-workload
  namespace: intel-power
spec:
  name: "performance-<my-node-name>-workload"
  allCores: true
  reservedCPUs:
  - 0
  - 1
  powerNodeSelector:
    kubernetes.io/hostname: <my-node-name>
  powerProfile: "performance"
```

**Comment by KingKoopasGoomba** [comment]
Hi,
this is not directly an issue with power manager, it is down to container runtime mounting sysfs as read only (rather than required read-write) in the power-agent pod
I would suggest updating you contaier runtime and/or kernel version

**Comment by LeiZhou-97** [comment]
Thanks for you advice.
My configuration is
**containerd: v1.6.7
kernel: 5.15.0
OS: RHEL 8.6**
I saw one same issue in containerd repo as well. So finally my work around is to mount /sys manually.
```
 volumeMounts:
    - mountPath: /sys
      name: sys
  volumes:
    - name: sys
      hostPath:
        path: /sys
```

Sorry, I have one more question. But this is related to power-optimization-library,  if it's inappropriate to ask here, I can create a new issue in that project.
**Problem:**
Why did I fail to set EPP value?
**Error log:**
failed to set EPP value for core 2: write /sys/devices/system/cpu/cpu2/cpufreq/energy_performance_preference: **device or resource busy**
```
# try to set epp on host, but failed as well
$ echo power | sudo tee energy_performance_preference
power
tee: energy_performance_preference: Device or resource busy
```
But when I change the config of **scaling_governor**, the value of **energy_performance_preference** also changes.

**Comment by KingKoopasGoomba** [comment]
Writing to EPP file fails most likely fails for you because of the hardware managed P-states (HWP) being enabled
To verify that's the case check if `/sys/devices/system/cpu/intel_pstate/hwp_dynamic_boost`  exists

You can disable HWP by adding `intel_pstate=no_hwp` kernel parameter

For more information see https://www.kernel.org/doc/html/v4.19/admin-guide/pm/intel_pstate.html#active-mode-with-hwp

We will soon add a note to the Readme, and include automated check the next release

**Comment by LeiZhou-97** [comment]
Hi, @KingKoopasGoomba 
HWP being enabled may also cause same problem. 
But I found the root cause of my problem was that I set my cpu governor -> "performance"
**Like the source code said, if the current policy is "performance", it will refuse to set EPP to any values different.**
```
/*
* To avoid confusion, refuse to set EPP to any values different
* from 0 (performance) if the current policy is "performance",
* because those values would be overridden.
*/
if (epp > 0 && cpu_data->policy == CPUFREQ_POLICY_PERFORMANCE)
	return -EBUSY;
```

For now, it runs successfully on my cluster. Thanks so much!

**Comment by patricia-cahill** [comment]
This issue was addressed and fixed in the latest release v2.1.0 
Any more issues please reach out to us, thanks
